tools/bin/make-filesystem-image

```diff
@@ -3,8 +3,23 @@
 set -euo pipefail
 
 IMG="$1"
+DOCKER_IMAGE_SIZE_IN_MB="$2"
+
+# We have to pick a fixed size in advance for the .img file we create, so base it on the size
+# of the original Docker image to avoid either wasting space or having the later tar extraction
+# step fail with out of disk space errors. The image will be mounted read-only at runtime, so
+# does not need free space for app files (separate mounts are used for those). The multiplier
+# here is to account for the 5-6% loss of usable space due to ext3 filesystem overhead, as well
+# as to ensure a few MB additional free space headroom.
+IMG_SIZE_IN_MB=$((DOCKER_IMAGE_SIZE_IN_MB * 107 / 100))
```

Comment:
Should this be a loop that keeps adding a bit of space up until a particular threshold, to avoid any potential trouble in the future?

Or are we happy with the fact that this is now run in CI, so we'd catch it early anyway?

Reply:
The image creation / docker export / extraction process are quite slow, so we really don't want to have to repeat them.

I tested multiple image sizes and the filesystem overhead seemed linear, so I don't expect any issues in the future. I also tested the failure mode if the image is too small, and its loud and correctly causes generation/the CI run to fail.